builtin/logical/cassandra/util.go

```diff
@@ -44,6 +46,22 @@ func createSession(cfg *sessionConfig, s logical.Storage) (*gocql.Session, error
 		clusterConfig.ProtoVersion = 2
 	}
 
+	if len(cfg.ConnectTimeout) != 0 {
```

Comment:
None of this will be necessary once you make the above change; merely set the timeout directly, since you will always have an integer number of seconds (the configured value or else the default).